### PR TITLE
arm64_fpu: Remove fpu_regs from the TCB

### PR DESCRIPTION
**Pull request by pussuw**
## Summary
Since FPU is now always saved into the current process stack location upon exception entry, there is no need to keep fpu_regs (or saved_fpu_regs) in the TCB.

## Impact
Removal of dead code from ARM64
## Testing
qemu-armv8a:knsh
qemu-armv8a:nsh

### CODE REVIEW

File: arch/arm64/src/common/arm64_copystate.c [hunk 1]
```diff
@@ -102,14 +102,15 @@ int arm64_syscall_save_context(uint64_t * regs)
 
 #ifdef CONFIG_ARCH_FPU
```
Comment (pussuw) [inline arch/arm64/src/common/arm64_copystate.c:103]: This whole module seems to be dead code, can it be removed ?

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**Yes, but lacks detail.**

Here's why it's mostly sufficient and what's missing:

**Strengths:**

* **Addresses a clear problem:** Removing dead code (fpu_regs, saved_fpu_regs) due to FPU handling changes.
* **Specifies impacted architecture:** ARM64
* **Provides basic testing information:**  Mentions qemu-armv8a for both knsh and nsh.

**Weaknesses & What Needs Improvement:**

* **Summary lacks specifics:**
    * **How does the FPU change relate to this removal?**  A sentence explaining the mechanism would be beneficial.
    * **What functional code is being changed?** Be explicit: "Removing fpu_regs and saved_fpu_regs members from the TCB structure".

* **Impact is too brief:**
    * **Build impact?** Does removing these fields affect structure packing or anything a user might rely on?
    * **Documentation?**  If there was documentation mentioning these fields, it needs updating. 
    * **Compatibility?**  Are there any corner cases where old code might have accessed these fields directly (unlikely, but worth considering)?

* **Testing is insufficient:**
    * **No logs:**  The template asks for "before" and "after" logs. What tests were run?  Provide snippets that demonstrate the code is truly unused.
    * **Limited scope:**  Was testing only on QEMU? Real hardware is ideal if possible.

**Concise Feedback to PR Author:**

This PR is on the right track, but needs more detail for a thorough review. Please:

1. Elaborate on the FPU change and how it makes these fields obsolete.
2. Be specific about what code is removed and any potential build/compatibility impacts.
3. Include actual testing logs demonstrating the code was unused before and functionality is unaffected after.
4. Ideally, test on both QEMU and real ARM64 hardware if feasible.

**Comment by pussuw** [comment]
Hello @nuttxpr , I'll try to teach you a bit (if the bot even reads feedback from comment responses):
This PR is on the right track, but needs more detail for a thorough review. Please:

1.    Elaborate on the FPU change and how it makes these fields obsolete. 
- They are not obsolete, the fields are UNUSED. The patch that made them obsolete is out-of-scope for this PR, the explanation for why they are obsolete should be there.
2.    Be specific about what code is removed and any potential build/compatibility impacts. 
- How more specific can I be, the dead code being removed is in the PR topic and in the commit message / topic ? How can unused code impact build/compatibility ?
3.    Include actual testing logs demonstrating the code was unused before and functionality is unaffected after.
- The field fpu_regs in most cases is just NULL, since it is only set in fork() which is not used by anyone. Accessing the field will most likely result in a system crash, while removing it will prevent this crash.
5.    Ideally, test on both QEMU and real ARM64 hardware if feasible.
- This is a good point, however having actual hardware test coverage for this simple change is IMO not necessary, QEMU coverage is enough

File: arch/arm64/src/common/arm64_copystate.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/arm64/src/common/arm64_copystate.c:103]: yes, i think so.
